snap/snapcraft.yaml

```
@@ -58,25 +65,34 @@ parts:
       fi
 
       /usr/bin/dotnet publish src/Dotnet.Installer.Console --output "${SNAPCRAFT_PART_INSTALL}" \
-        --configuration Release -r "${RUNTIME_RID}" -p:DebugSymbols=false -p:DebugType=none
+        --configuration Debug -r "${RUNTIME_RID}" -p:DebugSymbols=true -p:DebugType=full
       chmod 555 "${SNAPCRAFT_PART_INSTALL}/Dotnet.Installer.Console"
-      
-      craftctl default
     override-stage: |
       craftctl default
-      # jq files
-      patchelf --set-interpreter /snap/dotnet/current/lib64/ld-linux-x86-64.so.2 usr/bin/jq
-      patchelf --force-rpath --set-rpath /snap/dotnet/current/usr/lib/x86_64-linux-gnu usr/bin/jq
-      # .NET Installer files
-      patchelf --set-interpreter /snap/dotnet/current/lib64/ld-linux-x86-64.so.2 Dotnet.Installer.Console
-      patchelf --force-rpath --set-rpath \$ORIGIN/netcoredeps:/snap/dotnet/current/usr/lib/x86_64-linux-gnu Dotnet.Installer.Console
-    prime:
-      - usr/bin/jq
-      - usr/lib/*/libjq.so*
-      - usr/lib/*/libonig.so*
-      - Dotnet.Installer.Console
-      - Configuration/*
-      - Scripts/*
+
+      if [ "${CRAFT_ARCH_BUILD_FOR}" = "amd64" ]; then
+        LIB_PATH=lib/x86_64-linux-gnu
+        INTERPRETER="/snap/dotnet/current/lib64/ld-linux-x86-64.so.2"
+
+        patchelf --force-rpath --set-rpath \$ORIGIN/../${LIB_PATH} \
+          ${CRAFT_STAGE}/usr/bin/jq
+        patchelf --force-rpath --set-rpath \$ORIGIN/usr/${LIB_PATH} \
+          ${CRAFT_STAGE}/Dotnet.Installer.Console
+      elif [ "${CRAFT_ARCH_BUILD_FOR}" = "arm64" ]; then
+        LIB_PATH=lib/aarch64-linux-gnu
+        INTERPRETER="/snap/dotnet/current/lib/ld-linux-aarch64.so.1"
+
+        patchelf --force-rpath --set-rpath \$ORIGIN/../${LIB_PATH}:\$ORIGIN/../../${LIB_PATH} \
+          ${CRAFT_STAGE}/usr/bin/jq
+        patchelf --force-rpath --set-rpath \$ORIGIN/${LIB_PATH}:\$ORIGIN/usr/${LIB_PATH} \
+          ${CRAFT_STAGE}/Dotnet.Installer.Console
+      else
+        echo "Unknown architecture (${CRAFT_ARCH_BUILD_FOR})"
```

nitpick: The architecture is probably known, but unsupported.

```suggestion
        echo "Unsupported architecture (${CRAFT_ARCH_BUILD_FOR})"
```